src/allmydata/test/test_introducer.py

```diff
@@ -314,6 +316,64 @@ def test_duplicate(self):
 
 NICKNAME = u"n\u00EDickname-%s" # LATIN SMALL LETTER I WITH ACUTE
 
+def foolscapEndpointForPortNumber(portnum):
+    """
+    Create an endpoint that can be passed to ``Tub.listen``.
+
+    :param portnum: Either an integer port number indicating which TCP/IPv4
+        port number the endpoint should bind or ``None`` to automatically
+        allocate such a port number.
+
+    :return: A two-tuple of the integer port number allocated and a
+        Foolscap-compatible endpoint object.
+    """
+    if portnum is None:
+        from twisted.internet import reactor
+        from twisted.internet.interfaces import IReactorSocket
+        if IReactorSocket.providedBy(reactor):
```

Comment:
We could probably at least move the interface import to the top? (I'm guessing reactor is imported here "in case something was trying to set a custom reactor"..?)